ons_alpha/bulletins/models.py

```diff
@@ -91,6 +92,14 @@ class BulletinPage(BasePage):
         index.AutocompleteField("get_admin_display_title"),
     ]
 
+    def save(self, *args, **kwargs):
+        # Set the previous_version field for corrections
+        if self.revisions.exists():
+            latest_revision = self.revisions.order_by("-created_at").first()
+            if latest_revision and not self.updates:
+                self.updates.append({"previous_version": latest_revision.pk})
```

Comment:
Issue: Manually adding custom fields into a streamfield is a bad idea. You'll want to pull out the latest correction block and confirm it doesn't already have a previous version attached.